### PR TITLE
Service redirects

### LINKED ISSUE #531: Pass through cherrypy.HTTPRedirect exceptions
In `invoke_service`, some exceptions should not be caught by the handler (currently on line 462).  For instance, `cherrypy.HTTPRedirect` should be allowed to be raised so that when a return value with a Last-Modified header hasn't changed, the data need not be transferred to the caller.

Rather than `except:`, I'd recommend
```
except cherrypy.HTTPRedirect:
    raise
except Exception:
    tangelo.http_status(501, "Web Service Error")
```

There are other cherrypy exceptions that perhaps should also be raised.  Candidates are `cherrypy.HTTPError`, which would allow services to just raise an explicit error to exit early, `cherrypy.InternalRedirect`, which might be usable to redirect to another endpoint, and `cherrypy.NotFound`, which would allow a service to return an explicit 404 to exit early.

### PR DESCRIPTION
**Pull request by ronichoudhury**
Closes #531

### CODE REVIEW

File: tests/redirection.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+def test_redirect():
+    r = requests.get(fixture.url("redirect/redirect"))
+    assert r.ok
+    assert r.status_code == 200
+    assert r.text == "hello, world\n"
+
+    r = requests.get(fixture.url("redirect/internal_redirect"))
+    assert r.ok
+    assert r.status_code == 200
+    assert r.text == "hello, world\n"
```
Comment (manthey) [inline tests/redirection.py]: You should also test that the redirect sends a 303 and the internal_redirect doesn't, like so:

```
r = requests.get(fixture.url("redirect/redirect"), allow_redirects=False)
assert r.ok
assert r.status_code == 303
r = requests.get(fixture.url("redirect/internal_redirect"),
                 allow_redirects=False)
assert r.ok
assert r.status_code == 200
assert r.text == "hello, world\n"
````

**Comment by manthey** [comment]
A variation on this, rather than redirecting, is to serve a file internally.  This is done in cherrypy via 
`return cherrypy.lib.static.serve_file(<path>, content_type=<mime type>))` rather than returning with the data.  This allows serving files that *aren't* in the root tree, and therefore adds some functionality that doesn't exist in `internal_redirect`.  This feature could be added to this PR, or I could create a new issue for it.

File: tests/redirection.py (hunk 1 above)
Comment (ronichoudhury) [inline tests/redirection.py]: Ah, I did try this, but didn't know about ``allow_redirects``. Thanks!

**Comment by ronichoudhury** [comment]
The "file service" functionality is in a different PR that I'm working on now. Expect to see that sometime today. It has precisely the advantage you mentioned, that it allows for service of arbitrary file content that isn't otherwise available from Tangelo.

File: tests/redirection.py (hunk 1 above)
Comment (ronichoudhury) [inline tests/redirection.py]: @manthey, please take another look.

**Comment by manthey** [comment]
LGTM

**Pull request merged**